src/plugins/world_control/WorldControl.cc

```diff
@@ -219,11 +227,21 @@ void WorldControl::ProcessMsg()
 {
   std::lock_guard<std::recursive_mutex> lock(this->dataPtr->mutex);
 
-  if (!this->dataPtr->pause && this->dataPtr->msg.paused())
+  // If the pause state of the message doesn't match the pause state of this
+  // plugin, then play/pause must have occurred elsewhere (for example, the
+  // command line). If the pause state of the message matches the pause state
+  // of this plugin, but the pause state of the message differs from the
+  // previous message's pause state, this means that a pause/play request from
+  // this plugin has been registered by the server
+  if ((!this->dataPtr->pause && this->dataPtr->msg.paused()) ||
+      (!this->dataPtr->lastStatsMsgPaused && this->dataPtr->msg.paused()))
```

Comment:
Both these conditions have `&& this->dataPtr->msg.paused()` - can that be moved out to make it a bit simpler? For example

```
  if (this->dataPtr->msg.paused() &&
      (!this->dataPtr->pause || !this->dataPtr->lastStatsMsgPaused))
```

The same goes for the condition below. I'm having a hard time parsing it even with the comments.

Reply:
Thanks for the suggestion! I have applied it in https://github.com/ignitionrobotics/ign-gui/pull/306/commits/7ff16ce47ffbb1e0dc37d005333a4111a13e505d. Hopefully it's clearer now :crossed_fingers: :grimacing: